### PR TITLE
Appointment Booker V.1.0

### PR DESCRIPTION
**Pull request by 0sugo**
## What does this PR do?

This PR is raised to implement the following features: 👍 :

- [x] Create rails app using rails new
- [x] Setup controllers,routes and models (user, reservation, doctor)
- [x] Rendered data in JSON format 
- [x] Use GitFlow to introduce new features 💡  
- [x] Added validations and associations
- [x] Added API documentation using Swagger
- [x] Setup rubocop 

**LINK**: [Frontend](https://github.com/0sugo/appointment_booker_frontend/pull/15)

## Description of the tasks completed 👇 

- [x] Appointment Booker V.1.0
  
## How should it be tested

Clone the repository
```
git clone https://github.com/0sugo/appointment_booker_backend.git
```
Navigate to the folder
```
cd appointment_booker_backend
```
Checkout the branch
```
git checkout development
```
Run app

- Start the frontend server by following the above link
- Start backend server using the below command
```
rails s -p 4000
```

### CODE REVIEW

**Review by vichuge — COMMENTED** [review]
Hi dear student, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

cat_massage

## Invalid Code Review Request

You have submitted a project, with an issue: The pull request link that you should have sent should be opened from the dev branch into the main branch and include all the changes.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: README.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
+# Doctor_appointment_reservation_back_end
```
Comment (Meltrust) [inline README.md:1]: - [ ]  You are using Rspec, but the "test" folder still exists, which is used by another test framework. Kindly consider deleting that folder if you are not using it. 👍

**Review by Meltrust — CHANGES_REQUESTED** [review]
Hi @0sugo,

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation, but you are almost there!






### To highlight:

-  Tests are passing✔️
-  Nice code organization ✔️ 
-  Api is working well✔️ 
-  Good readme ✔️ 




#### You are really close to finishing the Microverse program!! Keep it up! 👍👍👍

You can do it

### After implementing the requested changes, please submit another review request. ♻️
_Check the comments under the review._







Cheers and Happy coding!👏👏👏

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the previous reviews unless it is requested otherwise._**

------

**Review by Meltrust — APPROVED** [review]
#### Hi @0sugo,


# **Wow, you did it 🎉**
Brilliant
## Thank you for the changes implemented 💪 🥇 ㊗️ 
#### _Unless you want to add more features, go ahead to your final presentation ⏩ ⏩ ⏩_ 
## You are about to finish the Microverse program.  You have come a long way!!!




### Good luck in the software industry!!  I'll see you there. ✨
### **Congratulations!!!!!!**  🎉
applause

### To highlight

- Superfluous test folder has been removed✔️ 
- When deleting a doctor, their reservations will be deleted✔️ 
- Great job✔️ 



Cheers and Happy coding!👏👏👏


------

**Pull request merged**